bambi/plots/plotting.py

```diff
@@ -0,0 +1,299 @@
+# pylint: disable = protected-access
+# pylint: disable = too-many-function-args
+# pylint: disable = too-many-nested-blocks
+from typing import Union
+import warnings
+
+import arviz as az
+from arviz.plots.backends.matplotlib import create_axes_grid
+from arviz.plots.plot_utils import default_grid
+import numpy as np
+from pandas.api.types import is_categorical_dtype, is_numeric_dtype, is_string_dtype
+
+from bambi.models import Model
+from bambi.plots.effects import comparisons, predictions
+from bambi.plots.plot_types import plot_categoric, plot_numeric
+from bambi.plots.utils import get_covariates, ConditionalInfo
+from bambi.utils import get_aliased_name, listify
+
+
+def plot_cap(
+    model: Model,
+    idata: az.InferenceData,
+    covariates: Union[str, list],
+    target: str = "mean",
+    pps: bool = False,
+    use_hdi: bool = True,
+    prob=None,
+    transforms=None,
+    legend: bool = True,
+    ax=None,
+    fig_kwargs=None,
+    subplot_kwargs=None,
+):
+    """Plot Conditional Adjusted Predictions
+
+    Parameters
+    ----------
+    model : bambi.Model
+        The model for which we want to plot the predictions.
+    idata : arviz.InferenceData
+        The InferenceData object that contains the samples from the posterior distribution of
+        the model.
+    covariates : list or dict
+        A sequence of between one and three names of variables in the model.
+    target : str
+        Which model parameter to plot. Defaults to 'mean'. Passing a parameter into target only
+        works when pps is False as the target may not be available in the posterior predictive
+        distribution.
+    pps: bool, optional
+        Whether to plot the posterior predictive samples. Defaults to ``False``.
+    use_hdi : bool, optional
+        Whether to compute the highest density interval (defaults to True) or the quantiles.
+    hdi_prob : float, optional
+        The probability for the credibility intervals. Must be between 0 and 1. Defaults to 0.94.
+        Changing the global variable ``az.rcParam["stats.hdi_prob"]`` affects this default.
+    legend : bool, optional
+        Whether to automatically include a legend in the plot. Defaults to ``True``.
+    transforms : dict, optional
+        Transformations that are applied to each of the variables being plotted. The keys are the
+        name of the variables, and the values are functions to be applied. Defaults to ``None``.
+    ax : matplotlib.axes._subplots.AxesSubplot, optional
+        A matplotlib axes object or a sequence of them. If None, this function instantiates a
+        new axes object. Defaults to ``None``.
+    subplot_kwargs : optional
+        Keyword arguments passed to the matplotlib subplot function as a dict. This allows you
```

Comment:
Actually, these are not passed to `plt.subplot()`, right? I would say these kwargs determine how different variables are mapped to different aspects of the visualization, without focusing on _how_ it happens internally.

Reply:
You are right; good catch.